src/nodes.coffee

```diff
@@ -2229,8 +2229,11 @@ exports.Assign = class Assign extends Base
             nestedProperties = prop.value.variable.base.properties
             [prop.value.value, nestedSourceDefault] = prop.value.value.cache o
           if nestedProperties
-            nestedSource = new Value source.base, source.properties.concat [new Access getPropKey prop]
-            nestedSource = new Value new Op '?', nestedSource, nestedSourceDefault if nestedSourceDefault
+            if source.properties
```

Comment:
Is this another case that would work better if whatever `source` is was wrapped in a `Value`? @helixbass 

Alternatively, should the case where there are no `properties` not *wrap* `source` in a `Value`? I can't really remember how this all works, but in the case that `source` has no `properties`, the next level of key won't be added (`source.properties.concat [new Access getPropKey prop]`). It must be working if the tests pass, but I don't really understand why...